PrivateHeaders/APPX/ZIP.h

```diff
@@ -327,6 +328,11 @@ namespace appx {
            << "xmlns=\"http://schemas.microsoft.com/appx/2010/blockmap\" "
            << "HashMethod=\"http://www.w3.org/2001/04/xmlenc#sha256\">";
         for (const ZIPFileEntry &entry : otherEntries) {
+            const std::string suffix = ".appx";
+            if (suffix.size() < entry.fileName.size() &&
+                    std::equal(suffix.rbegin(), suffix.rend(), entry.fileName.rbegin())) {
+                continue;
+            }
```

Comment:
Does including the .appx files in the block manifest do any harm? If so, I suggest calling this function without the .appx files in the file list. In other words, in the caller, make sure otherEntries doesn't include the .appx files.